lib/searches.js

```diff
@@ -179,48 +160,124 @@ let dfsHelper = function (url, edges, level) {
 // * A searchDepth of 0 terminates immediately.
 // * A searchDepth of n performs a search depth of n, beginning at the rootUrl
 //   and following random nodes until a depth of n is reached.
-let depthFS = function (rootUrl, searchDepth) {
+// Will return 1 for successful search, -1 for search that terminated early, and 2 
+// for search where the search term was found.
+let depthFS = function (rootUrl, searchDepth, cookie) {
     let currentDepth = 0;
     let edges = [];
     let nextUrl = rootUrl;
+    let linkError = false;
 
     let searchingPromise = new Promise((resolve, reject) => {
         console.log("Crawling level 0...");
 
         // Get random links until the searchDepth is reached.
         function getRandomLink(url) {
+            
+            // If searchDepth is reached, insert the tree into the database
             if (currentDepth >= searchDepth) {
-                resolve(edges);
+                resolve(1);
                 return;
-            }
-
+            } 
+            
+            // Call dfsHelper to insert the next random link into the edge array
             let helperPromise = dfsHelper(nextUrl, edges, currentDepth);
 
             helperPromise.then(() => {
                 // If link error, return results thus far.
                 if (edges[currentDepth] === undefined){
-                    resolve(edges);
+                    resolve(-1);
+                    return;
                 }
+               
+                if (edges[currentDepth].termFound === true){
+                    resolve(2);
+                    return;
+                }
+                // Insert link into temp edges table
+                let tempInsert = dbAPI.insertIntoTempEdges(edges[currentDepth].SourceUrl, edges[currentDepth].DestinationUrl, cookie, 'Ian Dalrymple', 'DFS', searchDepth, rootUrl);   
+               
+                tempInsert.then((result) => {});
```

Comment:
These callbacks are probably unnecessary; they don't do anything and we never use this promise elsewhere, so it looks like "fire and forget" is good enough.